app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -198,28 +233,104 @@ class TableVisualization extends Visualization {
             dataTruncated = parsedData.all_rows_count !== rowData.length
         }
 
-        // If the table contains more rows than an upper limit, the engine will send only some of all rows.
+        // Update Status Bar
+        this.createRowDropdown(
+            parsedData.all_rows_count === undefined ? 1 : parsedData.all_rows_count,
+            dataTruncated
+        )
+
         // If data is truncated, we cannot rely on sorting/filtering so will disable.
-        // A pinned row is added to tell the user the row count and that filter/sort is disabled.
-        const col_span = '__COL_SPAN__'
-        if (dataTruncated) {
-            columnDefs[0].colSpan = p => p.data[col_span] || 1
-        }
         this.agGridOptions.defaultColDef.filter = !dataTruncated
         this.agGridOptions.defaultColDef.sortable = !dataTruncated
         this.agGridOptions.api.setColumnDefs(columnDefs)
-        if (dataTruncated) {
-            const field = columnDefs[0].field
-            const extraRow = {
-                [field]: `Showing ${rowData.length} of ${parsedData.all_rows_count} rows. Sorting and filtering disabled.`,
-                [col_span]: columnDefs.length,
-            }
-            this.agGridOptions.api.setPinnedTopRowData([extraRow])
-        }
         this.agGridOptions.api.setRowData(rowData)
         this.updateTableSize(this.dom.getAttributeNS(null, 'width'))
     }
 
+    makeOption(value, label) {
+        const optionElem = document.createElement('option')
+        optionElem.value = value
+        optionElem.appendChild(document.createTextNode(label))
+        return optionElem
+    }
+
+    makeButton(label, onclick) {
+        const buttonElem = document.createElement('button')
+        buttonElem.name = label
+        buttonElem.appendChild(document.createTextNode(label))
+        buttonElem.addEventListener('click', onclick)
+        return buttonElem
+    }
+
+    createRowDropdown(all_rows_count, dataTruncated) {
```

Comment:
Can you add a description what is `RowDropdown`, what does it contain and where is displayed?